src/Core/Field.h

```diff
@@ -479,12 +479,20 @@ class Field
         return true;
     }
 
-    template <typename T> auto & safeGet() const
+    template <typename T> const auto & safeGet() const &
     {
         return const_cast<Field *>(this)->safeGet<T>();
     }
+    template <typename T> auto safeGet() const &&
+    {
+        return safeGet<T>();
+    }
 
-    template <typename T> auto & safeGet();
+    template <typename T> auto & safeGet() &;
+    template <typename T> auto safeGet() &&
+    {
+        return safeGet<T>();
```

Comment:
`std::move`?

Reply:
exactly!
thanks

Reply:
...though I'm not sure we use move semantics in any of ours Field types :)
but probably now we should - will look into it on next iteration :)